### ISSUE TITLE
Amazon FreeRTOS for Renesas MCUs versioning (release tags name)

### ISSUE THREAD

**Issue by HirokiIshiguro** [renesas/amazon-freertos#8]
**Is your feature request related to a problem? Please describe.**  
Current release tags naming is not good.
For example, we do not know reference Amazon FreeRTOS version from current release tags.

old discussion thread: https://github.com/renesas-rx/amazon-freertos/issues/15

**Describe the solution you would like.**  
Following scheme is good.

`<aws_version> - <device_name> - <renesas_version> `
`e.g. v1.4.9-rx-0.1.13, 201906.00_Major-rx-0.1.15, 201906.00_Major-rz-0.0.1`

**Additional context**  
None

**Comment by HirokiIshiguro** [comment]
We need to consider FreeRTOS (with IoT libraries) and FreeRTOS (Kernel only) like:

FreeRTOS(with IoT libraries)
<aws_version>-<device_name>-<renesas_version>
"201906.00_Major-rx-0.1.15"

FreeRTOS(Kernel Only)
<aws_version>-<device_name>-<renesas_version>
"V10.3.1-kernel-only-rx-0.1.15"

**Comment by HirokiIshiguro** [comment]
branch naming:
<device_name>-<feature>-<aws_version>
e.g. rx-development-202012.00

**Comment by HirokiIshiguro** [comment]
Original tools version naming:
<renesas_version>
e.g. 1.0.0

Original tools example: https://github.com/renesas/mot-file-converter

**Comment by HirokiIshiguro** [comment]
asset naming:
same as versioning.
<versioning>.zip
e.g. v202002.00-rx-1.0.0.zip

**Comment by HirokiIshiguro** [comment]
# versioning list for tags/branches
## Amazon FreeRTOS
### tags
FreeRTOS(with IoT libraries) : for new style (https://www.freertos.org/aws-reference-integrations.html -> copy https://github.com/renesas/iot-reference-rx)

Basically, Renesas would like to use LTS version like: https://github.com/FreeRTOS/FreeRTOS-LTS

`v<aws_version>-<device_name>-<renesas_version>-<release_candidate_indicator(option)>`
"v202210.01-LTS-rx-1.0.0-rc"
"v202210.01-LTS-rx-1.0.0"

<details>
<summary>deprecated</summary>

FreeRTOS(with IoT libraries) : for old style (deprecated: https://github.com/aws/amazon-freertos -> fork https://github.com/renesas/amazon-freertos)
`v<aws_version>-<device_name>-<renesas_version>-<release_candidate_indicator(option)>-<aws_lts_version(option)>`
"v202002.00-rx-1.0.0-rc"
"v202002.00-rx-1.0.0"
"v202012.00-rx-1.0.1-202012.01-LTS"

</details>

FreeRTOS(Kernel Only)
`<aws_version>-<device_name>-<renesas_version>-<release_candidate_indicator(option)>`
"V10.3.1-rx-1.0.0-rc"
"V10.3.1-rx-1.0.0"

### branches
`<device_name>-development-<aws_version>-<feature(option)>`
"rx-development-202012.00"
"rx-development-202012.00-cellular"
"rx-development-V10.4.4"

## Azure RTOS
### tags
We should follow ThreadX tag naming: https://github.com/azure-rtos/threadx/tags
This versioning is applied for whole of Azure RTOS versioning.

`<threadx_version>-<device_name>-<renesas_version>-<release_candidate_indicator(option)>`
"v6.1.9_rel-rx-1.0.0-rc"
"v6.1.9_rel-rx-1.0.0"

### branches
`<device_name>-development-<threadx_version>-<feature(option)>`
"rx-development-v6.1.9_rel"
"rx-development-v6.1.9_rel-cellular"

## Azure RTOS: Embedded-Wireless-Framework
### tags
We should follow Embedded-Wireless-Framework tag naming: https://github.com/Azure/embedded-wireless-framework/tags

`<ewf_version>+renesas.<serial_number>.<device_name(option)>-<release_candidate_indicator(option)>`
"1.0.0-preview.1+renesas.0"
"1.0.0-preview.1+renesas.0.rx"
"1.0.0-preview.2.renesas.rx+renesas.0.rx"

### branches
`<device_name>-development-<ewf_version>-<feature(option)>`
"rx-development-1.0.0-preview.1"
"rx-development-1.0.0-preview.1-ppp-support"

## Aruduino
### tags
We should follow Arduino Library tag naming: https://github.com/arduino/Arduino/tags

`<arduino_version>-<device_name>-<renesas_version>-<release_candidate_indicator(option)>`
"1.8.19-rl78-1.0.0-rc"
"1.8.19-rl78-1.0.0"

### branches
`<device_name>-development-<arduino_version>-<feature(option)>`
"rx-development-1.8.19"
"rx-development-1.8.19-sdcard"

## RX Driver Package
### tags
`<renesas_version>-<release_candidate_indicator(option)>`
"V1.32-rc"
"V1.32"

### branches
`exceptional_modules` -> The FIT Modules that is not included in official RX Driver Package, etc.
Others are now considering.

# common
## tags
### <renesas_version> rule
* Basically
  * We follow semantic versioning. https://semver.org/

### <release_candidate_indicator(option)> rule
* rc<sequence_number>
  * rc0 -> ok
  * rc1 -> ok
  * rc2 -> ok
  * rc10 -> ok
  * rc100 -> ok
  * rc.0 -> ng

## branches
`<device_name>-development-<feature>`
"rx-development-addingfuncA"

## notes
* We reset the version if <aws_version> would be changed.
  * For example:
    * v202002.00-rx-1.0.0 -> ok
    * v202002.00-rx-1.0.1 -> ok
    * v202002.00-rx-1.0.2 -> ok
    * v202012.00-rx-1.0.0 -> ok
    * v202012.00-rx-1.0.3 -> ng
